src/generate.dart

```diff
@@ -74,90 +85,133 @@ class DiagramGenerator {
 
   /// The temporary directory used to write screenshots and cropped out images
   /// into.
-  Directory tmpDir;
+  Directory temporaryDirectory;
 
-  static String get projectDir =>
-      path.joinAll(path.split(path.fromUri(Platform.script))..removeLast());
+  static String get projectDir {
+    return path.joinAll(path.split(path.fromUri(Platform.script))..removeLast());
+  }
 
   Future<Null> generateDiagram() async {
     await _collectScreenshot();
     await _optimizeImages();
     if (cleanup) {
-      await tmpDir.delete(recursive: true);
+      await temporaryDirectory.delete(recursive: true);
     }
   }
 
-  bool _captureScreenshot(String outputName) {
-    print('Capturing screenshot into $outputName.');
-    ProcessResult processResult = processRunner(flutterCommand, ['screenshot', '--out=$outputName'],
-        workingDirectory: projectDir);
-    if (processResult.exitCode != 0) {
-      print("Failed to run command '$flutterCommand screenshot --out=$outputName' in ${tmpDir.path}:\n${processResult.stderr}");
-      return false;
+  static const Duration timeout = const Duration(seconds: 60);
+
+  Future<List<String>> runFlutter({
+    @required List<String> args,
+    @required String until,
+    AsyncCallback then,
+  }) async {
+    final Process process = await processStarter(flutterCommand, args, workingDirectory: projectDir);
+    final List<String> lines = <String>[];
+    process.stdout.transform(UTF8.decoder).transform(const LineSplitter()).listen((String data) {
+      lines.add(data);
+    });
+    process.stderr.transform(UTF8.decoder).transform(const LineSplitter()).listen((String data) {
+      lines.add(data);
```

Comment:
Don't you want to separate stderr output from the stdout output, and only look on one for the "until" string?  Otherwise you might get interleaved blocks in the output.

Reply:
Only complete lines get added, so interleaving shouldn't really matter (and will help when the output is dumped to the console in case of error).